specification/security/resource-manager/common/v1/types.json

```diff
@@ -158,6 +158,91 @@
     "AscLocationProperties": {
       "type": "object",
       "description": "An empty set of properties"
+    },
+    "ResourceDetails": {
+      "type": "object",
+      "description": "Details of the resource that was assessed",
+      "discriminator": "source",
+      "properties": {
+        "source": {
+          "type": "string",
+          "description": "The platform where the assessed resource resides",
+          "enum": [
+            "Azure",
+            "OnPremise"
+          ],
+          "x-ms-enum": {
+            "name": "source",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Azure",
+                "description": "Resource is in Azure"
+              },
+              {
+                "value": "OnPremise",
+                "description": "Resource in an on premise machine connected to Azure cloud"
+              }
+            ]
+          }
+        }
+      },
+      "required": [
+        "source"
+      ]
+    },
+    "AzureResourceDetails": {
+      "type": "object",
+      "description": "Details of the Azure resource that was assessed",
+      "x-ms-discriminator-value": "Azure",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ResourceDetails"
+        }
+      ],
+      "properties": {
+        "id": {
```

Comment:
This is implied by the resource ID of the assessment.  I am not against having it inside the assessment as well as long as you respond to moves of the parent resource and update this property accordingly

Reply:
we are responding to a resource move
we added it so the object will be consistent even if the resource type is not Azure (AWS\OnPrem etc.) since they cannot be implied from the resource ID